api/room/schema.py

```diff
@@ -188,8 +188,13 @@ class Arguments:
 
     @Auth.user_roles('Admin')
     def mutate(self, info, room_id, **kwargs):
+        if 'structure_id' in kwargs:
+            structure_id = kwargs.get('structure_id').strip()
+            if not structure_id:
+                del kwargs['structure_id']
+            else:
+                validate_structure_id(**kwargs)
```

Comment:
Why do you validate structure_id yet you have deleted it from kwargs in line 194?

Reply:
I only validate if it's not an empty string if line 194 is run, line 196 will never run